### PR TITLE
Fix use struct `AutoScalingConfigData`

### PR DESCRIPTION
**Pull request by shiimaxx**
Used `AutoScalingConfigData` with some place of replaceable.

@netmarkjp 
Please review.

### CODE REVIEW

File: halib/config.go [hunk 1]
```diff
@@ -15,22 +15,16 @@ type MetricConfig struct {
...
-	AutoScaling      bool     `yaml:"autoscaling,omitempty" json:"autoscaling,omitempty"`
-	AutoScalingCount int      `yaml:"autoscaling_count,omitempty" json:"autoscaling_count,omitempty"`
-	HostPrefix       string   `yaml:"host_prefix,omitempty" json:"host_prefix,omitempty"`
-	Proxies          []string `yaml:"proxies" json:"proxies"`
-	Disabled         bool     `yaml:"disabled,omitempty" json:"disabled,omitempty"`
+	GroupName   string                `yaml:"group_name" json:"group_name"`
+	IP          string                `yaml:"ip" json:"ip"`
+	Hostname    string                `yaml:"hostname" json:"hostname"`
+	Port        int                   `yaml:"port" json:"port"`
+	AutoScaling AutoScalingConfigData `yamo:"autoscaling,omitempty" json:"autoscaling,omitempty"`
```
Comment (netmarkjp) [inline halib/config.go]: maybe typo. `s/yamo/yaml/` :stuck_out_tongue_winking_eye:

File: halib/struct.go [hunk 2]
```diff
@@ -38,9 +38,9 @@ type AutoScalingData struct {
 
 // AutoScalingConfigData is actual autoscaling config data
 type AutoScalingConfigData struct {
```
Comment (netmarkjp) [inline halib/struct.go]: At `AutoScalingGroupName` and `AutoScalingCount`, remmoving JSON tag was not necessary I think.

Comment (shiimaxx) [reply halib/struct.go]: I added YAML tag in `AutoScalingConfigData` but didn't remove JSON tag.
What does "removing JSON tag" refer to?

**Comment by shiimaxx** [comment]
@netmarkjp 
Thanks for review!
I fixed code and commented.

File: halib/struct.go (hunk 2 above)
Comment (netmarkjp) [reply halib/struct.go]: oh, sorry... It's my mis-reading. You're right.

**Comment by netmarkjp** [comment]
@shiimaxx Thanks, looks good :relaxed:

**Pull request merged**